### PR TITLE
 [tests-only][full-ci] Added tests for searching groups by a normal user

### PR DESCRIPTION
**Pull request by ghost**
## Description

Adding test to get user data as a normal user using search queries
```cucumber
Scenario Outline: non-admin user search for a group
Scenario Outline: non-admin user tries to search for a group with less than 3 characters
Scenario Outline: non-admin user tries to search for a group with invalid characters/token
```
and this PR also fixes some of the incorrect json assertion in file `tests/acceptance/features/apiGraph/getGroup.feature`

## Related Issue

- Part of: https://github.com/owncloud/ocis/issues/7944
- https://github.com/owncloud/ocis/issues/8115

## How Has This Been Tested?

- test environment:
- test case 1:
- test case 2:
- ...

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

**Review by saw-jan — DISMISSED** [review]
LGTM

File: tests/TestHelpers/GraphHelper.php [hunk 1]
```diff
@@ -554,6 +554,32 @@ public static function getGroup(
 		);
 	}
 
+	/**
+	 * @param string $baseUrl
+	 * @param string $xRequestId
+	 * @param string $adminUser
+	 * @param string $adminPassword
```
Comment (grgprarup) [inline tests/TestHelpers/GraphHelper.php:561]:
```suggestion
	 * @param string $user
	 * @param string $password
```
Do we need an admin user to search?

File: tests/TestHelpers/GraphHelper.php [hunk 2]
```diff
@@ -554,6 +554,32 @@ public static function getGroup(
...
+	 * @param string $searchTerm
+	 *
+	 * @return ResponseInterface
+	 */
+	public static function searchGroup(
+		string $baseUrl,
+		string $xRequestId,
+		string $adminUser,
+		string $adminPassword,
```
Comment (grgprarup) [inline tests/TestHelpers/GraphHelper.php:570]:
```suggestion
		string $user,
		string $password,
```

File: tests/acceptance/features/apiGraph/getGroup.feature [hunk 3]
```diff
@@ -444,3 +452,117 @@ Feature: get groups and their members
...
+              "type": "string",
+              "enum": ["Token '<token>' is invalid"]
+            }
+          }
+        }
+      }
+    }
+    """
+    Examples:
+    | groupName  | token   |
```
Comment (grgprarup) [inline tests/acceptance/features/apiGraph/getGroup.feature:566]:
```suggestion
    | group-name  | token   |
```
We are using kebab case for example table header.

File: tests/acceptance/features/apiGraph/getGroup.feature [hunk 4]
```diff
@@ -444,3 +452,117 @@ Feature: get groups and their members
...
+            }
+          }
+        }
+      }
+    }
+    """
+    Examples:
+      | group |
+      | tea   |
+      | %22tea-%22  |
```
Comment (saw-jan) [inline tests/acceptance/features/apiGraph/getGroup.feature:500]:
```suggestion
      | group      |
      | tea        |
      | %22tea-%22 |
```

File: tests/acceptance/features/apiGraph/getGroup.feature [hunk 5]
```diff
@@ -444,3 +452,117 @@ Feature: get groups and their members
...
+            }
+          }
+        }
+      }
+    }
+    """
+    Examples:
+    | group-name  | token   |
+    | tea-lovers  | -lovers |
+    | tea@lovers  | @lovers |
```
Comment (saw-jan) [inline tests/acceptance/features/apiGraph/getGroup.feature:568]:
```suggestion
    | group      | token   |
    | tea-lovers | -lovers |
    | tea@lovers | @lovers |
```

**Review by saw-jan — COMMENTED** [review]
let's make it consistent

File: tests/acceptance/features/apiGraph/getGroup.feature [hunk 6]
```diff
@@ -444,3 +452,117 @@ Feature: get groups and their members
...
+            }
+          }
+        }
+      }
+    }
+    """
+    Examples:
+      | group       |
+      | tea         |
+      | %22tea-%22  |
```
Comment (saw-jan) [inline tests/acceptance/features/apiGraph/getGroup.feature:500]:
```suggestion
      | group      |
      | tea        |
      | %22tea-%22 |
```

File: tests/acceptance/features/bootstrap/GraphContext.php [hunk 7]
```diff
@@ -1887,6 +1887,28 @@ public function userGetsDetailsOfTheGroupUsingTheGraphApi(string $user, string $
...
+	/**
+	 * @When user :user tries to search for group :searchTerm using Graph API
+	 * @When user :user searches for group :searchTerm using Graph API
+	 *
+	 * @param string $user
+	 * @param string $groupName
+	 *
+	 * @return void
+	 */
+	public function userSearchesForGroupUsingGraphApi($user, $groupName):void {
```
Comment (saw-jan) [inline tests/acceptance/features/bootstrap/GraphContext.php:1899]:
```suggestion
	 * @param string $searchTerm
	 *
	 * @return void
	 */
	public function userSearchesForGroupUsingGraphApi($user, $searchTerm):void {
```

**Review by saw-jan — DISMISSED** [review]
:+1:

**Review by SagarGi — APPROVED** [review]
LGTM :+1:  Please make CI green

**Review by saw-jan — APPROVED** [review]

**Pull request merged**